hoodie-client/src/main/java/com/uber/hoodie/index/bloom/HoodieBloomIndex.java

```diff
@@ -390,17 +390,14 @@ private boolean shouldCompareWithFile(BloomIndexFileInfo indexInfo, String recor
     // Here as the recordRDD might have more data than rowKeyRDD (some rowKeys' fileId is null),
     // so we do left outer join.
     return rowKeyRecordPairRDD.leftOuterJoin(rowKeyFilenamePairRDD).values().map(v1 -> {
-      HoodieRecord<T> record = v1._1();
+      // When you have a record in multiple files in the same partition, then rowKeyRecordPairRDD will have 2
+      // entries with the same exact in memory copy of the HoodieRecord and the 2 separate filenames that the
+      // record is found in. This will result in setting currentLocation 2 times and it will fail the second time.
+      // So creating a new in memory copy of the hoodie record.
+      HoodieRecord<T> record = new HoodieRecord<>(v1._1());
```

Comment:
Instead of checking if the currentLocation is set, I think creating the record every time is cleaner.

Reply:
What is the implication of doing this ? Will this trigger a huge number of objects being created ?